druid-shell/src/platform/x11/window.rs

```diff
@@ -720,13 +743,15 @@ impl Window {
             .idle_pixmaps
             .last()
             .ok_or_else(|| anyhow!("after rendering, no pixmap to present"))?;
+        let scale = self.get_scale()?;
         if let Some(present) = borrow_mut!(self.present_data)?.as_mut() {
-            present.present(self.app.connection(), pixmap, self.id, &invalid)?;
+            present.present(self.app.connection(), pixmap, self.id, &invalid, scale)?;
             buffers.idle_pixmaps.pop();
         } else {
-            for r in invalid.rects() {
-                let (x, y) = (r.x0 as i16, r.y0 as i16);
-                let (w, h) = (r.width() as u16, r.height() as u16);
+            for rect in invalid.rects() {
+                let rect = rect.to_px(scale);
```

Comment:
I think you need to round this rectangle (for the same reason as explained below)